### PR TITLE
Storybook source

### PR DESCRIPTION
**Pull request by mark-tate**

### CODE REVIEW

File: packages/source-storybook/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "types": "./dist/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./dist/index.d.ts",
+      "import": "./dist/index.js",
+      "node": "./dist/index.js"
+    }
+  },
+  "engines": {
+    "node": ">= 16.0.0"
```
Comment (DavieReid) [inline packages/source-storybook/package.json:18]: most packages have `  "node": ">=16.10.0 || >=18.0.0"`

File: packages/source-storybook/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      },
+      content: ``
+    };
+    return [...result, storyPage];
+  }, []);
+};
+
+const StorybookSource: Source<StorybookSourceOptions> = {
+  create(options, { schedule }): Observable<StorybookPage[]> {
+    // @ts-ignore
```
Comment (DavieReid) [inline packages/source-storybook/src/index.ts:83]: ignore without disabling eslint will prob fail linting....whats being ignored?

File: packages/source-storybook/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      agent: proxyEndpoint ? HttpsProxyAgent(proxyEndpoint) : undefined,
+      timeout: requestTimeout,
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+
+    return timer(schedule.initialDelayMs, delayMs).pipe(
+      switchMap(() => {
+        console.log('checking for new Storybook patterns', delayMs, schedule, options);
```
Comment (DavieReid) [inline packages/source-storybook/src/index.ts]: We prefix log statements with `[Mosaic]` for easy tracking

File: packages/source-storybook/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  prefixDir: z.string(),
+  proxyEndpoint: z.string().url().optional(),
+  stories: z.array(
+    z.object({
+      description: z.string(),
+      url: z.string(),
+      additionalData: z.object({}).passthrough().optional(),
+      additionalTags: z.array(z.string()).optional(),
+      filter: z
+        .any({ required_error: 'A regex is required to filter stories' })
```
Comment (DavieReid) [inline packages/source-storybook/src/index.ts:22]: required error on an optional?

Comment (mark-tate) [reply packages/source-storybook/src/index.ts:22]: ah yes... it was required and then I added filterTags as well... good shout!
Will remove

File: packages/source-storybook/package.json (hunk 1 above)
Comment (mark-tate) [reply packages/source-storybook/package.json:18]: done

File: packages/source-storybook/src/index.ts (hunk 3 above)
Comment (mark-tate) [reply packages/source-storybook/src/index.ts]: done

File: packages/source-storybook/src/index.ts (hunk 2 above)
Comment (mark-tate) [reply packages/source-storybook/src/index.ts:86]: added with a comment linking to issue

Comment (mark-tate) [reply packages/source-storybook/src/index.ts]: was caused by multiple versions of zod... fixed versions and error went away

**Review by DavieReid — APPROVED** [review]

**Pull request closed without merging**